### ISSUE TITLE
automated perf/accessibility checks in deploy pipeline CI/CD

### ISSUE THREAD

**Issue by aaronhans** [cagov/Alpha#37]
Use this tool https://github.com/treosh/lighthouse-ci-action as part of our build/deploy process

Right now there is an azure pipeline process which is deploying every branch to staging. That could be modified to only deploy the staging branch to staging, then this github action can run and audit the site there. Going to be some discovery and learning during this process and we should distribute the knowledge gained throughout the team.

High level goal:
Use an automated tool which generates reports against a running version of the site using branch which we want to deploy and gives us:
- Performance measurements
- Accessibility measurements

Regressions in these scores should block promotion to production
Scores should be able to be tracked over time, ideally reports from each build would be viewable with the cute score dial graphics

**Comment by aaronhans** [comment]
I feel like this is a big task so putting all the devs as participants, it is not incredibly urgent, probably want to iron out azure pipeline first maybe this tool integrates there better than github actions. Anybody who is deeply interested in this should feel free to take ownership of this. If nobody is that into it I'll own this ticket and rely on the other devs for support.

**Comment by abquirarte** [comment]
This done, yeah? (@aaronhans )

**Comment by aaronhans** [comment]
Not done yet, Chad is working on it @abquirarte 

> This done, yeah? (@aaronhans )

**Comment by aaronhans** [comment]
Made Chad sole owner of this because he has already done some work on it. 

One way to accomplish running these tests is to run them against a local version of the site which would be built inside the docker container created by the github action.

This might also be the best way to run the automated e2e test suite. I'll be taking ownership of integrating the e2e test suite in another ticket and we can learn from each other's work

**Comment by aaronhans** [comment]
Went through a few iterations of this process and came up with the following requirements:

- [ ] should not noticeably slow down the build process
- [ ] should run regularly
- [ ] results of performance and accessibility tests should be stored so we can review trends
- [ ] show results in graphs we can showoff
- [ ] should allow budget setting and trigger alerts when thresholds are breached

We accomplished many of these after iterating on a solution integrated into the build process but did cause significant slowdown of the build testing run so we went with a SAAS solution from calibreapp.com which provides all of these features. It is possible to build this ourselves (calibre is using the open source lighthouse CLI tool) but it will take several days of engineering work to match the base featureset and we don't necessarily want to manage this platform ourselves. Got approval from @KimberlyglennTSM to get the $60/month plan for this service covered. We are currently running on a 2 week free trial which will expire 2/21.

**Comment by aaronhans** [comment]
I think we should close this issue now, I am really happy with calibre. If we continue to like using the tool after we sign up for the paid plan we should integrate this feature which will mark our graphs with deployment markers so it will be extremely obvious where a regression occurred. This isn't important immediately but would be useful when we have a larger team with a lot more code movement: https://calibreapp.com/docs/features/pull-request-reviews#create-a-pull-request

**Comment by aaronhans** [comment]
<img width="733" alt="Screen Shot 2020-02-07 at 7 28 43 AM" src="https://user-images.githubusercontent.com/353360/74042025-a2d69b80-497b-11ea-98da-7a5a469f8c4e.png">

**Comment by abquirarte** [comment]
Closing! ✅
